### ISSUE TITLE
Model Trapheus as an AWS Systems Manager automation

### ISSUE THREAD

**Issue by namitad** [intuit/Trapheus#174]
Trapheus is currently modelled as an AWS step function or state machine.

This issue looks to alternatively support Trapheus as a SSM automation which can be run on AWS, cloned and shared with other accounts etc.
https://docs.aws.amazon.com/systems-manager/latest/userguide/documents-creating-content.html

**Comment by stationeros** [comment]
What is the outcome expected in terms of deliverable here ?

**Comment by namitad** [comment]
the outcome expected here is a json/yaml with the automation document - which can also be modelled as a cloud formation template

**Comment by stationeros** [comment]
What is the limitation in the way its currently modelled?

**Comment by stationeros** [comment]
did you take a look at https://github.com/aws-samples/ssm-for-rds-snapshot-cross-region-copy

**Comment by namitad** [comment]
yes. there is also this for cross account https://docs.aws.amazon.com/systems-manager-automation-runbooks/latest/userguide/automation-aws-sharerdssnapshot.html provided by amazon.

these are still some parts of what trapheus solves. i feel it will be good to have a possible end to end automation of trapheus on SSM.

**Comment by stationeros** [comment]
I am saying the above looks like a solved problem to me with https://github.com/aws-samples/ssm-for-rds-snapshot-cross-region-copy. If you feel the output of this issue would be any different from https://github.com/aws-samples/ssm-for-rds-snapshot-cross-region-copy ,  feel free to convert this issue into a discussion and create few fine grained issues.

**Comment by namitad** [comment]
https://github.com/aws-samples/ssm-for-rds-snapshot-cross-region-copy - this solves only for copying Aurora RDS snapshots across regions - so the output here is a copy of a aurora snapshot in different regions. Trapheus currently does creation of snapshots and automated db restoration  for all RDS albeit in one account - the output here is a restored RDS instance/cluster for any kind of RDS. Modelling as a ssm could help enable that complete restoration flow to be executed more seamlessly in one account, across accounts and different regions without the requirement of setting up the CFT in every region or account.

**Comment by stationeros** [comment]
Got it, if thats the case convert this issue into a discussion , and create more detailed and focussed issues, which can be solved in HF timeframe and can encourage a contribution, since this would involve multiple things. See if you can break this into smaller subtasks if possible.
